src/MBC_UserImport_Source_Niche.php

```diff
@@ -254,209 +195,270 @@ public function setter($message)
    */
   public function process()
   {
-
-    $payload = $this->addCommonPayload($this->importUser);
-    $existing['log-type'] = 'user-import-niche';
-    $existing['source'] = $payload['source'];
-
-    // Add welcome email details to payload
-    $this->addWelcomeEmailSettings($this->importUser, $payload);
-
-    // Check for existing email account in MailChimp
-    $subscribed = $this->mbcUserImportToolbox->checkExistingEmail(
-      $this->importUser,
-      $existing
-    );
-    if (!$subscribed) {
-      $this->addEmailSubscriptionSettings($this->importUser, $payload);
+    // Shortcuts.
+    $northstar = &$this->northstar;
+    $input = &$this->user;
+
+    // User status variables.
+    $userIsNew = false;
+    $userIsNewToCampaign = false;
+
+    // Lookup on Northstar by email.
+    $identityByEmail = $northstar->getUser('email', $input['email']);
+    if (!empty($input['mobile'])) {
+      $identityByMobile = $northstar->getUser('mobile', $input['mobile']);
+    } else {
+      $identityByMobile = false;
     }
 
-    // Drupal user
-    $this->mbcUserImportToolbox->checkExistingDrupal(
-      $this->importUser,
-      $existing
-    );
-    if (empty($existing['drupal-uid'])) {
-      $importUser = (object) $this->importUser;
-      // Set user registration source.
-      $importUser->source = 'niche';
-
-      // Lookup user on Northstar.
-      $northstarUser = $this->mbToolbox->lookupNorthstarUser($importUser);
-      if ($northstarUser && !empty($northstarUser->drupal_id)) {
-        // User is missing from phoenix, but present on Northstar.
-        // Sync credentials:
-        $importUser->email = $northstarUser->email;
-        $importUser->mobile = $northstarUser->mobile;
-      }
+    // Process all possible cases and merge data based on identity load results:
+    if (empty($identityByEmail) && empty($identityByMobile)) {
+      // ****** New user ******
+      $userIsNew = true;
+      $userIsNewToCampaign = true;
 
-      // Trigger user creation on Northstar, will force-create user
-      // user on Phoenix.
-      $northstarUser = $this->mbToolbox->createNorthstarUser($importUser);
+      // Create Northstar and Phoenix accounts.
+      self::log(
+        'User not found, creating new user on Northstar: %s',
+        json_encode($input)
+      );
+      $identity = $northstar->createUser($input);
+    } elseif (!empty($identityByEmail) && empty($identityByMobile)) {
+      // ****** Existing user: only email record exists ******
+      $identity = &$identityByEmail;
```

Comment:
`$identity` was not a shortcut, but a `let`-kind-of variable. Passing it by reference: yes, this was not necessary, just a micro-optimization.